src/include/network/lan/hardware.rb

```diff
@@ -829,7 +828,7 @@ def storeHW(_key, _event)
         end
         if LanItems.type == "vlan"
           # for vlan devices named vlanN pre-set vlan_id to N, otherwise default to 0
-          LanItems.vlan_id = "#{nm["vlan".size].to_i}"
+          LanItems.vlan_id = nm[VLAN_SIZE].to_i.to_s
```

Comment:
when already touching this ```LanItems.vlan_id = nm[VLAN_SIZE..-1]``` looks better in my POV. At least it should remove issue when vlan id is longer than one char, which is currently present.